taichi/program/sparse_matrix.cpp

```diff
@@ -24,6 +24,12 @@
         }                                                                      \
   }
 
+#define INSTANTIATE_SPMV(type, storage)                               \
+  template void                                                       \
+  EigenSparseMatrix<Eigen::SparseMatrix<type, Eigen::storage>>::spmv( \
+      taichi::lang::Program *prog, const taichi::lang::Ndarray &x,    \
```

Comment:
Why is `x` const but not `y`?

Reply:
`y` is used to store the computed results of `A @ x = y`. `x` is the input of sparse matrix-vector multiplication.

Reply:
Hmm but you are not modifying the `Ndarray` object. You are just writing to the memory address `y` wraps.